test/python/test_mlir_qiskit_translation.py

```
@@ -0,0 +1,423 @@
+
+class RejectQiskit(importlib.abc.MetaPathFinder):
+    def find_spec(self, fullname, path, target=None):
+        if fullname == "qiskit" or fullname.startswith("qiskit."):
+            raise AssertionError("MLIR binding attempted to import Qiskit")
+        return None
+
+sys.meta_path.insert(0, RejectQiskit())
+import mqt.core.mlir
+assert "qiskit" not in sys.modules
+"""
+    subprocess.run([sys.executable, "-c", script], check=True)  # ruff: ignore[subprocess-without-shell-equals-true]
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)